instrumentation/akka-context-propagation-2.5/src/main/java/io/opentelemetry/instrumentation/auto/akkaconcurrent/AkkaForkJoinPoolInstrumentation.java

```diff
@@ -28,27 +28,33 @@
 import io.opentelemetry.instrumentation.auto.api.InstrumentationContext;
 import io.opentelemetry.instrumentation.auto.api.concurrent.ExecutorInstrumentationUtils;
 import io.opentelemetry.instrumentation.auto.api.concurrent.State;
-import io.opentelemetry.instrumentation.auto.javaconcurrent.AbstractExecutorInstrumentation;
 import io.opentelemetry.javaagent.tooling.Instrumenter;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.concurrent.Executor;
 import net.bytebuddy.asm.Advice;
 import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
 import net.bytebuddy.matcher.ElementMatcher;
 
 @AutoService(Instrumenter.class)
-public final class AkkaExecutorInstrumentation extends AbstractExecutorInstrumentation {
+public final class AkkaForkJoinPoolInstrumentation extends Instrumenter.Default {
 
-  public AkkaExecutorInstrumentation() {
-    super(AbstractExecutorInstrumentation.EXEC_NAME + ".akka_fork_join");
+  public AkkaForkJoinPoolInstrumentation() {
+    super("akka_context_propagation");
   }
 
   @Override
   protected boolean defaultEnabled() {
     return false;
   }
 
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    // This might need to be an extendsClass matcher...
```

Comment:
What's this comment referring to? Is this a safe change?

Reply:
This class used to extend `AbstractExecutorInstrumentation` whose `typeMatcher()` matches all Executors in `WHITELISTED_EXECUTORS`.

But this instrumentation is specific to `akka.dispatch.forkjoin.ForkJoinPool`, so this is an optimization.

Probably it should use `extendsClass`, since the class it applies to is not final, in case someone subclasses it and overrides the method(s) that we instrument.

In general, I don't think we are super consistent on when we use `named(..)` vs `extends(named(..))`, I opened #1097 to tack this.